### PR TITLE
ci: add semantic release and github actions

### PR DESCRIPTION
**Pull request by Zalaras**
Add semantic release and GitHub actions

### Description
- Added Github Actions
- Added semantic release
- Updated PR template and Readme

### Issue
[PRSDM-5464](https://spandigital.atlassian.net/browse/PRSDM-5464)

### Checklist before merging

* [ ] Did you test your changes locally?
* [ ] Did you update the CHANGELOG?
* [ ] Is the documentation updated for this change? (If Required)

### CODE REVIEW

**Review by CharlRitterDev — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        with:
+          persist-credentials: false # to disable default GITHUB_TOKEN
+      
+      - name: Semantic Release
+        uses: cycjimmy/semantic-release-action@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.PAT }}
+        with:
+          extra_plugins: |
+            - @semantic-release/exec
```
Comment (meyerhp) [inline .github/workflows/release.yml:21]: Doesn't look like this is used anymore

**Review by meyerhp — APPROVED** [review]
Looks good overall. One small issue below

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false # to disable default GITHUB_TOKEN
+      
+      - name: Semantic Release
+        uses: cycjimmy/semantic-release-action@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.PAT }}
```
Comment (meyerhp) [inline .github/workflows/release.yml:18]: I don't think this is needed. The github release plugin works with the default token afk

**Review by meyerhp — APPROVED** [review]

**Review by kmorake — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (Zalaras) [reply .github/workflows/release.yml:21]: oops thanks

**Pull request merged**